R/as_draws_array.R

```diff
@@ -150,10 +150,10 @@ is_draws_array_like <- function(x) {
 
 #' @export
 `[.draws_array` <- function(x, i, j, ..., drop = FALSE) {
-  # TODO: add a warning that 'drop' is ignored?
+  # TODO: add a warning that 'drop' can lead to also dropping the class?
   # TODO: allow for argument 'reserved' as in '[.draws_df'
   #   right now this fails because NextMethod() cannot ignore arguments
-  out <- NextMethod("[", drop = FALSE)
+  out <- NextMethod("[")
   class(out) <- class(x)
```

Comment:
@paul-buerkner I guess the reason this doesn't work yet is because it's still setting the class to draws_array before exiting the function. However, the logic you used to get around this with draws_df doesn't quite work here since we don't have the "reserved" argument. Ideas?